homeassistant/components/homeassistant_hardware/silabs_multiprotocol_addon.py

```diff
@@ -306,6 +323,71 @@ async def _async_install_addon(self) -> None:
                 self.flow_manager.async_configure(flow_id=self.flow_id)
             )
 
+    async def _async_uninstall_addon(
+        self, addon_manager: AddonManager, *, continue_flow: bool = True
+    ) -> None:
+        """Uninstall an add-on."""
+        try:
+            try:
+                info = await addon_manager.async_get_addon_info()
+            except AddonError:
+                info = None
+
+            # Do not try to uninstall an addon if it is already uninstalled
+            if info is not None and info.state == AddonState.NOT_INSTALLED:
+                return
+
+            await addon_manager.async_uninstall_addon()
+            await self._async_wait_until_addon_state(
+                addon_manager, AddonState.NOT_INSTALLED
+            )
+        finally:
+            if continue_flow:
+                # Continue the flow after show progress when the task is done.
+                self.hass.async_create_task(
+                    self.flow_manager.async_configure(flow_id=self.flow_id)
+                )
```

Comment:
I think it would be nicer to split the function in two, one which uninstalls an addon, and one which first waits for an addon to uninstall and then resumes the flow.